components/server/src/workspace/workspace-service.ts

```diff
@@ -437,10 +437,21 @@ export class WorkspaceService {
         user: User,
         workspaceId: string,
         options: StartWorkspaceOptions = {},
+        restrictToRegular = true,
     ): Promise<StartWorkspaceResult> {
         await this.auth.checkPermissionOnWorkspace(user.id, "start", workspaceId);
 
-        const workspace = await this.doGetWorkspace(user.id, workspaceId);
+        const { workspace, latestInstance } = await this.getWorkspace(user.id, workspaceId);
```

Comment:
This was the only missing functionality compared to GitpodServerImpl.startWorkspace, which was missed during earlier refactorings.